diff -rupN libelf-0.8.13/config.sub originals/libelf-0.8.13/config.sub
--- libelf-0.8.13/config.sub	2020-05-10 14:58:48.043368832 +0000
+++ originals/libelf-0.8.13/config.sub	2005-05-21 15:28:37.000000000 +0000
@@ -1167,7 +1167,7 @@ case $os in
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
-	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* | -btos*)
+	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN libelf-0.8.13/configure originals/libelf-0.8.13/configure
--- libelf-0.8.13/configure	2020-05-10 14:58:48.045368851 +0000
+++ originals/libelf-0.8.13/configure	2008-05-23 08:18:03.000000000 +0000
@@ -3445,14 +3445,6 @@ echo "$ac_t""$mr_cv_target_elf" 1>&6
 	LINK_SHLIB='$(LD) -G -z text -h $(SONAME)'
 	INSTALL_SHLIB='$(INSTALL_PROGRAM)'
 	;;
-	  i686-pc-btos*)
-	PICFLAGS='-fPIC -DPIC'
-	SHLINK_SFX='.ell'
-	SONAME_SFX='.ell'
-	LINK_SHLIB='$(CC) -shared -Wl,-soname,$(SONAME)'
-	INSTALL_SHLIB='$(INSTALL_PROGRAM)'
-	DEPSHLIBS='-lc'
-	;;
       *)
 	echo "configure: warning: shared libraries not supported for $host" 1>&2
 	mr_enable_shared=no