diff -rupN libiconv-1.14/build-aux/config.sub originals/libiconv-1.14/build-aux/config.sub
--- libiconv-1.14/build-aux/config.sub	2017-06-15 22:50:21.020993430 +0100
+++ originals/libiconv-1.14/build-aux/config.sub	2009-06-21 12:17:33.000000000 +0100
@@ -1282,7 +1282,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -btos*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN libiconv-1.14/configure originals/libiconv-1.14/configure
--- libiconv-1.14/configure	2017-06-15 22:50:21.032993418 +0100
+++ originals/libiconv-1.14/configure	2011-08-07 18:49:02.000000000 +0100
@@ -12455,16 +12455,6 @@ uts4*)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   ;;
-  
-btos*)
-  libname_spec='$name'
-  shrext_cmds=".ell"
-  need_lib_prefix=no
-  library_names_spec='$libname${shared_ext}'
-  dynamic_linker="$host_os ELOADER"
-  shlibpath_var=LIBPATH
-  soname_spec='$libname${shared_ext}'
-  ;;
 
 *)
   dynamic_linker=no
diff -rupN libiconv-1.14/lib/Makefile.in originals/libiconv-1.14/lib/Makefile.in
--- libiconv-1.14/lib/Makefile.in	2017-06-15 23:07:27.140086101 +0100
+++ originals/libiconv-1.14/lib/Makefile.in	2011-08-07 18:48:03.000000000 +0100
@@ -25,7 +25,7 @@ DEFS = -DLIBDIR=\"$(libdir)\" -DBUILDING
 -DENABLE_RELOCATABLE=1 -DIN_LIBRARY -DINSTALLDIR=\"$(libdir)\" -DNO_XMALLOC \
 -Dset_relocation_prefix=libiconv_set_relocation_prefix \
 -Drelocate=libiconv_relocate @DEFS@
-LIBTOOL = @LIBTOOL@ --tag=CC
+LIBTOOL = @LIBTOOL@
 LIBTOOL_COMPILE = $(LIBTOOL) --mode=compile
 LIBTOOL_LINK = $(LIBTOOL) --mode=link
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install
diff -rupN libiconv-1.14/libcharset/build-aux/config.sub originals/libiconv-1.14/libcharset/build-aux/config.sub
--- libiconv-1.14/libcharset/build-aux/config.sub	2017-06-15 22:50:21.036993415 +0100
+++ originals/libiconv-1.14/libcharset/build-aux/config.sub	2009-06-21 12:17:33.000000000 +0100
@@ -1214,7 +1214,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -btos* )
+	      | -skyos* | -haiku* | -rdos*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN libiconv-1.14/src/Makefile.in originals/libiconv-1.14/src/Makefile.in
--- libiconv-1.14/src/Makefile.in	2017-06-15 23:09:26.671986827 +0100
+++ originals/libiconv-1.14/src/Makefile.in	2011-06-04 18:05:58.000000000 +0100
@@ -22,7 +22,7 @@ CFLAGS = @CFLAGS@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 INCLUDES = -I. -I$(srcdir) -I.. -I../include -I$(srcdir)/../include -I../srclib -I$(srcdir)/../srclib
-LIBTOOL = @LIBTOOL@ --tag=CC
+LIBTOOL = @LIBTOOL@
 LIBTOOL_COMPILE = $(LIBTOOL) --mode=compile
 LIBTOOL_LINK = $(LIBTOOL) --mode=link
 LIBTOOL_INSTALL = $(LIBTOOL) --mode=install
diff -rupN libiconv-1.14/srclib/error.c originals/libiconv-1.14/srclib/error.c
--- libiconv-1.14/srclib/error.c	2017-06-15 22:50:21.040993412 +0100
+++ originals/libiconv-1.14/srclib/error.c	2011-08-07 14:42:06.000000000 +0100
@@ -49,10 +49,6 @@
 # define _(String) String
 #endif
 
-#ifdef __btos__
-#include <btos.h>
-#endif
-
 /* If NULL, error will flush stdout, then print on stderr the program
    name, a colon and a space.  Otherwise, error will call this
    function without parameters instead.  */
@@ -131,8 +127,6 @@ is_open (int fd)
      that they are open but point to an INVALID_HANDLE_VALUE.  There is no
      fcntl, and the gnulib replacement fcntl does not support F_GETFL.  */
   return (HANDLE) _get_osfhandle (fd) != INVALID_HANDLE_VALUE;
-# elif defined(__btos__)
-  return bt_queryhandle(btos_get_handle(fd));
 # else
 #  ifndef F_GETFL
 #   error Please port fcntl to your platform
