diff -rupN grub-2.00/build-aux/config.sub originals/grub-2.00/build-aux/config.sub
--- grub-2.00/build-aux/config.sub	2016-03-02 19:12:57.342274671 +0000
+++ originals/grub-2.00/build-aux/config.sub	2012-06-18 22:08:46.000000000 +0100
@@ -1368,8 +1368,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-              | -btos* )
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN grub-2.00/config.h originals/grub-2.00/config.h
--- grub-2.00/config.h	1970-01-01 01:00:00.000000000 +0100
+++ originals/grub-2.00/config.h	2012-06-27 22:11:28.000000000 +0100
@@ -0,0 +1,56 @@
+#undef _LARGEFILE_SOURCE
+#undef _FILE_OFFSET_BITS
+#define _LARGEFILE_SOURCE
+#define _FILE_OFFSET_BITS 64
+#if defined(__PPC__) && !defined(__powerpc__)
+#define __powerpc__ 1
+#endif
+#if defined (GRUB_UTIL) || !defined (GRUB_MACHINE)
+#include <config-util.h>
+#define NESTED_FUNC_ATTR
+#else
+/* Define if C symbols get an underscore after compilation. */
+#define HAVE_ASM_USCORE 0
+/* Define it to \"addr32\" or \"addr32;\" to make GAS happy.  */
+#define ADDR32 addr32
+/* Define it to \"data32\" or \"data32;\" to make GAS happy. */
+#define DATA32 data32
+/* Define it to one of __bss_start, edata and _edata.  */
+#define BSS_START_SYMBOL __bss_start
+/* Define it to either end or _end.  */
+#define END_SYMBOL end
+/* Name of package.  */
+#define PACKAGE "grub"
+/* Version number of package.  */
+#define VERSION "2.00"
+/* Define to the full name and version of this package. */
+#define PACKAGE_STRING "GRUB 2.00"
+/* Define to the version of this package. */
+#define PACKAGE_VERSION "2.00"
+/* Define to the full name of this package. */
+#define PACKAGE_NAME "GRUB"
+/* Define to the address where bug reports for this package should be sent. */
+#define PACKAGE_BUGREPORT "bug-grub@gnu.org"
+/* Default boot directory name" */
+#define GRUB_BOOT_DIR_NAME "boot"
+/* Default grub directory name */
+#define GRUB_DIR_NAME "grub"
+/* Define to 1 if GCC generates calls to __enable_execute_stack().  */
+#define NEED_ENABLE_EXECUTE_STACK 0
+/* Define to 1 if GCC generates calls to __register_frame_info().  */
+#define NEED_REGISTER_FRAME_INFO 0
+/* Define to 1 to enable disk cache statistics.  */
+#define DISK_CACHE_STATS 0
+
+#define GRUB_TARGET_CPU "i386"
+#define GRUB_PLATFORM "pc"
+
+#define RE_ENABLE_I18N 1
+
+#if defined(__i386__)
+#define NESTED_FUNC_ATTR __attribute__ ((__regparm__ (1)))
+#else
+#define NESTED_FUNC_ATTR
+#endif
+
+#endif
diff -rupN grub-2.00/configure originals/grub-2.00/configure
--- grub-2.00/configure	2016-03-02 19:12:59.194274638 +0000
+++ originals/grub-2.00/configure	2012-06-26 12:59:16.000000000 +0100
@@ -9139,7 +9139,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -9185,7 +9185,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -9209,7 +9209,7 @@ rm -f core conftest.err conftest.$ac_obj
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -9254,7 +9254,7 @@ else
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
@@ -9278,7 +9278,7 @@ rm -f core conftest.err conftest.$ac_obj
     We can't simply define LARGE_OFF_T to be 9223372036854775807,
     since some C++ compilers masquerading as C compilers
     incorrectly reject 9223372036854775807.  */
-#define LARGE_OFF_T ((((off_t) 1 << 31) << 31) - 1 + (((off_t) 1 << 31) << 31))
+#define LARGE_OFF_T (((off_t) 1 << 62) - 1 + ((off_t) 1 << 62))
   int off_t_is_large[(LARGE_OFF_T % 2147483629 == 721
 		       && LARGE_OFF_T % 2147483647 == 1)
 		      ? 1 : -1];
diff -rupN grub-2.00/grub-core/disk/ahci.c originals/grub-2.00/grub-core/disk/ahci.c
--- grub-2.00/grub-core/disk/ahci.c	2016-03-02 19:12:57.346274671 +0000
+++ originals/grub-2.00/grub-core/disk/ahci.c	2012-03-10 12:23:16.000000000 +0000
@@ -600,8 +600,8 @@ grub_ahci_readwrite_real (struct grub_ah
 		dev->command_table[0].prdt[0].data_base,
 		dev->command_table[0].prdt[0].unused,
 		dev->command_table[0].prdt[0].size,
-		(unsigned)((char *) &dev->command_table[0].prdt[0]
-		- (char *) &dev->command_table[0]));
+		(char *) &dev->command_table[0].prdt[0]
+		- (char *) &dev->command_table[0]);
 
   if (parms->write)
     grub_memcpy ((char *) grub_dma_get_virt (bufc), parms->buffer, parms->size);
diff -rupN grub-2.00/grub-core/gnulib/sleep.c originals/grub-2.00/grub-core/gnulib/sleep.c
--- grub-2.00/grub-core/gnulib/sleep.c	2016-03-02 19:12:57.346274671 +0000
+++ originals/grub-2.00/grub-core/gnulib/sleep.c	2010-12-01 14:45:43.000000000 +0000
@@ -68,15 +68,6 @@ rpl_sleep (unsigned int seconds)
   return sleep (seconds);
 }
 
-#elif defined(__btos__)
-
-#include <dev/rtc.h>
-
-unsigned int sleep (unsigned int seconds){
-	bt_rtc_sleep(seconds * 1000);
-	return 0;
-}
-
 #else /* !HAVE_SLEEP */
 
  #error "Please port gnulib sleep.c to your platform, possibly using usleep() or select(), then report this to bug-gnulib."
diff -rupN grub-2.00/grub-core/kern/emu/hostdisk.c originals/grub-2.00/grub-core/kern/emu/hostdisk.c
--- grub-2.00/grub-core/kern/emu/hostdisk.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/kern/emu/hostdisk.c	2012-06-25 09:32:04.000000000 +0100
@@ -115,11 +115,6 @@ struct hd_geometry
 # endif /* ! RAW_FLOPPY_MAJOR */
 #endif /* defined(__NetBSD__) */
 
-#if defined(__btos__)
-#include <btos.h>
-#include <crt_support.h>
-#endif
-
 static struct
 {
   char *drive;
@@ -252,7 +247,6 @@ grub_util_get_fd_size (int fd, const cha
   struct disklabel label;
 # elif defined (__sun__)
   struct dk_minfo minfo;
-# elif defined (__btos__)
 # else
   unsigned long long nr;
 # endif
@@ -327,20 +321,6 @@ grub_util_get_fd_size (int fd, const cha
 
  fail:
 
-#elif defined(__btos__)
-
-  sector_size = 512;
-  log_sector_size = 9;
-
-  if (log_secsize) *log_secsize = 9;
-
-  bt_handle_t fh = btos_get_handle(fd);
-  bt_filesize_t pos = bt_fseek(fh, 0, FS_Relative);
-  bt_filesize_t size = bt_fseek(fh, 0, FS_Backwards);
-  bt_fseek(fh, pos, FS_Set);
-
-  return size;
-
   /* In GNU/Hurd, stat() will return the right size.  */
 #elif !defined (__GNU__)
 # warning "No special routine to get the size of a block device is implemented for your OS. This is not possibly fatal."
diff -rupN grub-2.00/grub-core/kern/emu/mm.c originals/grub-2.00/grub-core/kern/emu/mm.c
--- grub-2.00/grub-core/kern/emu/mm.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/kern/emu/mm.c	2012-02-07 23:13:14.000000000 +0000
@@ -25,10 +25,6 @@
 #include <string.h>
 #include <grub/i18n.h>
 
-#ifdef __btos__
-#include <malloc.h>
-#endif
-
 void *
 grub_malloc (grub_size_t size)
 {
diff -rupN grub-2.00/grub-core/lib/backtrace.c originals/grub-2.00/grub-core/lib/backtrace.c
--- grub-2.00/grub-core/lib/backtrace.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/lib/backtrace.c	2012-02-26 19:56:25.000000000 +0000
@@ -39,7 +39,7 @@ grub_backtrace_print_address (void *addr
 	  + segment->size > (grub_uint8_t *) addr)
 	{
 	  grub_printf ("%s.%x+%" PRIxGRUB_SIZE, mod->name, segment->section,
-		       (unsigned)((grub_uint8_t *) addr - (grub_uint8_t *) segment->addr));
+		       (grub_uint8_t *) addr - (grub_uint8_t *) segment->addr);
 	  return;
 	}
   }
diff -rupN grub-2.00/grub-core/lib/minilzo/lzoconf.h originals/grub-2.00/grub-core/lib/minilzo/lzoconf.h
--- grub-2.00/grub-core/lib/minilzo/lzoconf.h	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/lib/minilzo/lzoconf.h	2011-10-14 18:10:35.000000000 +0100
@@ -53,7 +53,7 @@
 #  include <config.h>
 #endif
 #include <limits.h>
-#include <stdint.h>
+#include <stddef.h>
 
 
 /***********************************************************************
diff -rupN grub-2.00/grub-core/lib/minilzo/minilzo.c originals/grub-2.00/grub-core/lib/minilzo/minilzo.c
--- grub-2.00/grub-core/lib/minilzo/minilzo.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/lib/minilzo/minilzo.c	2012-02-29 14:04:03.000000000 +0000
@@ -58,12 +58,7 @@
 #  include <config.h>
 #endif
 #include <limits.h>
-#include <stdint.h>
-#include <sys/types.h>
-#ifndef NULL
-#define NULL 0
-#endif
-typedef __PTRDIFF_TYPE__ ptrdiff_t;
+#include <stddef.h>
 #if defined(MINILZO_CFG_USE_INTERNAL_LZODEFS)
 
 #ifndef __LZODEFS_H_INCLUDED
diff -rupN grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c originals/grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c
--- grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c	2010-12-01 14:45:43.000000000 +0000
@@ -24,8 +24,6 @@
 #include "xz_private.h"
 #include "xz_lzma2.h"
 
-#pragma GCC diagnostic ignored "-Wattributes"
-
 /*
  * Range decoder initialization eats the first five bytes of each LZMA chunk.
  */
diff -rupN grub-2.00/grub-core/net/ip.c originals/grub-2.00/grub-core/net/ip.c
--- grub-2.00/grub-core/net/ip.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/net/ip.c	2012-06-09 10:12:46.000000000 +0100
@@ -403,7 +403,7 @@ grub_net_recv_ip4_packets (struct grub_n
 					    * sizeof (grub_uint32_t)))
     {
       grub_dprintf ("net", "IP packet too short: %" PRIdGRUB_SSIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    (nb->tail - nb->data));
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
@@ -654,7 +654,7 @@ grub_net_recv_ip6_packets (struct grub_n
   if (nb->tail - nb->data < (grub_ssize_t) sizeof (*iph))
     {
       grub_dprintf ("net", "IP packet too short: %" PRIdGRUB_SSIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/net/tcp.c originals/grub-2.00/grub-core/net/tcp.c
--- grub-2.00/grub-core/net/tcp.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/net/tcp.c	2012-06-22 16:42:07.000000000 +0100
@@ -766,7 +766,7 @@ grub_net_recv_tcp_packet (struct grub_ne
 					     >> 12) * sizeof (grub_uint32_t)))
     {
       grub_dprintf ("net", "TCP packet too short: %" PRIuGRUB_SIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/net/udp.c originals/grub-2.00/grub-core/net/udp.c
--- grub-2.00/grub-core/net/udp.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/net/udp.c	2012-06-09 10:12:46.000000000 +0100
@@ -155,7 +155,7 @@ grub_net_recv_udp_packet (struct grub_ne
   if (nb->tail - nb->data < (grub_ssize_t) sizeof (*udph))
     {
       grub_dprintf ("net", "UDP packet too short: %" PRIuGRUB_SIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/rs_decoder.S originals/grub-2.00/grub-core/rs_decoder.S
--- grub-2.00/grub-core/rs_decoder.S	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/grub-core/rs_decoder.S	1970-01-01 01:00:00.000000000 +0100
@@ -1,484 +0,0 @@
-	.file	"reed_solomon.c"
-	.text
-	.type	pol_evaluate, @function
-pol_evaluate:
-	pushl	%ebp
-	movl	%esp, %ebp
-	pushl	%edi
-	movl	%ecx, %edi
-	pushl	%esi
-	movl	%eax, %esi
-	pushl	%ebx
-	xorl	%eax, %eax
-	xorl	%ebx, %ebx
-.L2:
-	testl	%edx, %edx
-	js	.L11
-	movzbl	(%esi,%edx), %ecx
-	testb	%cl, %cl
-	je	.L3
-	movzbl	1049088(%ecx), %ecx
-	xorb	1048576(%ebx,%ecx), %al
-.L3:
-	addl	%edi, %ebx
-	cmpl	$255, %ebx
-	leal	-255(%ebx), %ecx
-	cmovge	%ecx, %ebx
-	decl	%edx
-	jmp	.L2
-.L11:
-	popl	%ebx
-	popl	%esi
-	popl	%edi
-	popl	%ebp
-	ret
-	.size	pol_evaluate, .-pol_evaluate
-	.type	gf_mul, @function
-gf_mul:
-	pushl	%ebp
-	testl	%eax, %eax
-	movl	%esp, %ebp
-	je	.L14
-	testl	%edx, %edx
-	je	.L14
-	movzbl	1049088(%eax), %ecx
-	movzbl	1049088(%edx), %eax
-	movb	1048576(%ecx,%eax), %al
-	jmp	.L13
-.L14:
-	xorl	%eax, %eax
-.L13:
-	popl	%ebp
-	ret
-	.size	gf_mul, .-gf_mul
-	.type	gauss_solve.constprop.2, @function
-gauss_solve.constprop.2:
-	pushl	%ebp
-	movl	%esp, %ebp
-	pushl	%edi
-	pushl	%esi
-	pushl	%ebx
-	subl	$40, %esp
-	movl	%eax, -24(%ebp)
-	xorl	%eax, %eax
-	movl	%edx, -16(%ebp)
-	movl	%ecx, -32(%ebp)
-.L17:
-	cmpl	-24(%ebp), %eax
-	jge	.L46
-	movl	$-1, 1049344(,%eax,4)
-	incl	%eax
-	jmp	.L17
-.L46:
-	xorl	%eax, %eax
-.L19:
-	cmpl	-16(%ebp), %eax
-	jge	.L47
-	movl	-32(%ebp), %edx
-	movb	$0, (%edx,%eax)
-	incl	%eax
-	jmp	.L19
-.L47:
-	movl	-16(%ebp), %eax
-	xorl	%edi, %edi
-	movl	$0, -20(%ebp)
-	incl	%eax
-	movl	%eax, -28(%ebp)
-	movl	%eax, -48(%ebp)
-.L21:
-	movl	-24(%ebp), %eax
-	cmpl	%eax, -20(%ebp)
-	jge	.L34
-	xorl	%ebx, %ebx
-.L35:
-	cmpl	-16(%ebp), %ebx
-	jl	.L22
-.L26:
-	cmpl	-16(%ebp), %ebx
-	je	.L27
-	movl	-20(%ebp), %eax
-	movl	$255, %edx
-	xorl	%esi, %esi
-	movl	%ebx, 1049344(,%eax,4)
-	movzbl	1052672(%ebx,%edi), %eax
-	movzbl	1049088(%eax), %eax
-	subl	%eax, %edx
-	movzbl	1048576(%edx), %eax
-	movl	%eax, -36(%ebp)
-	jmp	.L28
-.L22:
-	cmpb	$0, 1052672(%ebx,%edi)
-	jne	.L26
-	incl	%ebx
-	jmp	.L35
-.L28:
-	cmpl	%esi, -16(%ebp)
-	jl	.L48
-	movzbl	1052672(%esi,%edi), %eax
-	incl	%esi
-	movl	-36(%ebp), %edx
-	call	gf_mul
-	movb	%al, 1052671(%esi,%edi)
-	jmp	.L28
-.L48:
-	movl	-20(%ebp), %eax
-	incl	%eax
-	movl	%eax, -36(%ebp)
-	movl	-48(%ebp), %eax
-	leal	(%eax,%edi), %ecx
-.L30:
-	movl	-24(%ebp), %eax
-	cmpl	%eax, -36(%ebp)
-	je	.L27
-	movzbl	1052672(%ebx,%ecx), %eax
-	xorl	%esi, %esi
-	movl	%eax, -44(%ebp)
-.L31:
-	cmpl	%esi, -16(%ebp)
-	jl	.L49
-	movb	1052672(%esi,%ecx), %al
-	incl	%esi
-	movl	%ecx, -52(%ebp)
-	movl	-44(%ebp), %edx
-	movb	%al, -37(%ebp)
-	movzbl	1052671(%esi,%edi), %eax
-	call	gf_mul
-	movl	-52(%ebp), %ecx
-	xorb	-37(%ebp), %al
-	movb	%al, 1052671(%esi,%ecx)
-	jmp	.L31
-.L49:
-	incl	-36(%ebp)
-	addl	-28(%ebp), %ecx
-	jmp	.L30
-.L27:
-	incl	-20(%ebp)
-	addl	-28(%ebp), %edi
-	jmp	.L21
-.L34:
-	movl	-24(%ebp), %ebx
-	movl	-28(%ebp), %edi
-	movl	-16(%ebp), %eax
-	decl	%ebx
-	imull	%ebx, %edi
-	notl	%eax
-	movl	%eax, -24(%ebp)
-.L36:
-	testl	%ebx, %ebx
-	js	.L50
-	movl	1049344(,%ebx,4), %eax
-	movl	%eax, -20(%ebp)
-	incl	%eax
-	je	.L37
-	xorl	%ecx, %ecx
-	xorl	%esi, %esi
-.L38:
-	cmpl	-16(%ebp), %esi
-	jge	.L51
-	movl	-32(%ebp), %eax
-	movl	%ecx, -28(%ebp)
-	movzbl	(%eax,%esi), %edx
-	incl	%esi
-	movzbl	1052671(%esi,%edi), %eax
-	call	gf_mul
-	movl	-28(%ebp), %ecx
-	xorl	%eax, %ecx
-	jmp	.L38
-.L51:
-	movl	-16(%ebp), %eax
-	movl	-20(%ebp), %edx
-	xorb	1052672(%edi,%eax), %cl
-	movl	-32(%ebp), %eax
-	movb	%cl, (%eax,%edx)
-.L37:
-	decl	%ebx
-	addl	-24(%ebp), %edi
-	jmp	.L36
-.L50:
-	addl	$40, %esp
-	popl	%ebx
-	popl	%esi
-	popl	%edi
-	popl	%ebp
-	ret
-	.size	gauss_solve.constprop.2, .-gauss_solve.constprop.2
-	.globl	grub_reed_solomon_recover
-	.type	grub_reed_solomon_recover, @function
-grub_reed_solomon_recover:
-	pushl	%ebp
-	movl	%esp, %ebp
-	pushl	%edi
-	pushl	%esi
-	pushl	%ebx
-	subl	$64, %esp
-	testl	%ecx, %ecx
-	movl	%eax, -52(%ebp)
-	movl	%edx, -24(%ebp)
-	movl	%ecx, -44(%ebp)
-	je	.L52
-	movb	$0, 1049088
-	movb	$1, %al
-	xorl	%edx, %edx
-.L58:
-	movzbl	%al, %ecx
-	testb	%al, %al
-	movb	%al, 1048576(%edx)
-	movb	%al, 1048831(%edx)
-	movb	%dl, 1049088(%ecx)
-	jns	.L54
-	leal	(%ecx,%ecx), %eax
-	xorl	$29, %eax
-	jmp	.L55
-.L54:
-	addl	%eax, %eax
-.L55:
-	incl	%edx
-	cmpl	$255, %edx
-	jne	.L58
-	movl	-52(%ebp), %eax
-	addl	-24(%ebp), %eax
-	movl	%eax, -56(%ebp)
-.L57:
-	cmpl	$0, -24(%ebp)
-	je	.L52
-	movl	-44(%ebp), %ecx
-	addl	-24(%ebp), %ecx
-	cmpl	$102400, %ecx
-	jbe	.L94
-	imull	$200, -24(%ebp), %eax
-	xorl	%edx, %edx
-	divl	%ecx
-	xorl	%edx, %edx
-	movl	%eax, -28(%ebp)
-	imull	$200, -44(%ebp), %eax
-	sall	$9, -28(%ebp)
-	divl	%ecx
-	movl	%eax, -36(%ebp)
-	sall	$9, -36(%ebp)
-	jmp	.L59
-.L94:
-	movl	-44(%ebp), %eax
-	movl	%eax, -36(%ebp)
-	movl	-24(%ebp), %eax
-	movl	%eax, -28(%ebp)
-.L59:
-	movl	-28(%ebp), %eax
-	movl	-52(%ebp), %esi
-	leal	511(%eax), %ebx
-	movl	%ebx, -40(%ebp)
-	movl	-56(%ebp), %ebx
-	leal	-1(%eax), %edi
-	movl	%esi, -48(%ebp)
-	movl	%edi, -64(%ebp)
-	leal	511(%ebx,%eax), %eax
-	movl	%eax, -72(%ebp)
-.L91:
-	movl	-40(%ebp), %ebx
-	movl	%ebx, %eax
-	shrl	$9, %eax
-	movl	%eax, -20(%ebp)
-	movl	-36(%ebp), %eax
-	subl	-28(%ebp), %eax
-	addl	%ebx, %eax
-	shrl	$9, %eax
-	movl	%eax, -16(%ebp)
-	je	.L60
-	xorl	%eax, %eax
-	cmpl	$0, -20(%ebp)
-	jne	.L61
-.L60:
-	decl	-40(%ebp)
-	movl	-64(%ebp), %eax
-	incl	-48(%ebp)
-	cmpl	%eax, -40(%ebp)
-	jne	.L91
-	movl	-28(%ebp), %eax
-	movl	-36(%ebp), %edi
-	addl	%eax, -52(%ebp)
-	addl	%edi, -56(%ebp)
-	subl	%eax, -24(%ebp)
-	subl	%edi, -44(%ebp)
-	jmp	.L57
-.L61:
-	cmpl	-20(%ebp), %eax
-	jge	.L97
-	movl	-48(%ebp), %esi
-	movl	%eax, %edx
-	incl	%eax
-	sall	$9, %edx
-	movb	(%esi,%edx), %dl
-	movb	%dl, 1052159(%eax)
-	jmp	.L61
-.L97:
-	xorl	%eax, %eax
-.L65:
-	cmpl	-16(%ebp), %eax
-	jge	.L98
-	movl	-72(%ebp), %ecx
-	movl	%eax, %edx
-	incl	%eax
-	subl	-40(%ebp), %ecx
-	sall	$9, %edx
-	movl	-20(%ebp), %edi
-	movb	(%ecx,%edx), %dl
-	movb	%dl, 1052159(%eax,%edi)
-	jmp	.L65
-.L98:
-	movl	-16(%ebp), %eax
-	xorl	%esi, %esi
-	movl	-20(%ebp), %edi
-	movl	%eax, %ebx
-	shrl	%ebx
-	leal	-1(%edi,%eax), %eax
-	movl	%ebx, -32(%ebp)
-	movl	%eax, -60(%ebp)
-.L67:
-	movl	-60(%ebp), %edx
-	movl	%esi, %ecx
-	movl	$1052160, %eax
-	incl	%esi
-	call	pol_evaluate
-	movb	%al, 1051903(%esi)
-	cmpl	-16(%ebp), %esi
-	jl	.L67
-	xorl	%eax, %eax
-.L70:
-	cmpb	$0, 1051904(%eax)
-	jne	.L69
-	incl	%eax
-	cmpl	-16(%ebp), %eax
-	jl	.L70
-.L69:
-	cmpl	-16(%ebp), %eax
-	je	.L71
-	movl	-32(%ebp), %eax
-	xorl	%ecx, %ecx
-	xorl	%edx, %edx
-	leal	1(%eax), %esi
-	jmp	.L72
-.L71:
-	xorl	%eax, %eax
-	jmp	.L73
-.L77:
-	movb	1051904(%eax,%edx), %bl
-	incl	%eax
-	movb	%bl, 1052671(%eax,%ecx)
-	cmpl	%eax, -32(%ebp)
-	jge	.L77
-	incl	%edx
-	addl	%esi, %ecx
-.L72:
-	xorl	%eax, %eax
-	cmpl	-32(%ebp), %edx
-	jl	.L77
-.L76:
-	cmpl	-32(%ebp), %eax
-	jge	.L99
-	movb	$0, 1050368(%eax)
-	incl	%eax
-	jmp	.L76
-.L99:
-	movl	-32(%ebp), %ebx
-	movl	$1050368, %ecx
-	xorl	%esi, %esi
-	xorl	%edi, %edi
-	movl	%ebx, %eax
-	movl	%ebx, %edx
-	call	gauss_solve.constprop.2
-	movl	-16(%ebp), %eax
-	decl	%ebx
-	addl	-20(%ebp), %eax
-	movl	%eax, -32(%ebp)
-.L79:
-	cmpl	-32(%ebp), %esi
-	jge	.L100
-	movl	$255, %ecx
-	movl	%ebx, %edx
-	subl	%esi, %ecx
-	movl	$1050368, %eax
-	call	pol_evaluate
-	cmpb	1048576(%esi), %al
-	jne	.L80
-	movb	%al, 1050624(%edi)
-	movl	-60(%ebp), %eax
-	movl	%edi, %edx
-	incl	%edi
-	subl	%esi, %eax
-	movl	%eax, 1050880(,%edx,4)
-.L80:
-	incl	%esi
-	jmp	.L79
-.L100:
-	xorl	%eax, %eax
-.L82:
-	cmpl	%edi, %eax
-	jge	.L101
-	movb	$1, 1052672(%eax)
-	incl	%eax
-	jmp	.L82
-.L101:
-	movb	1051904, %al
-	xorl	%esi, %esi
-	movl	$1, %ecx
-	movb	%al, 1052672(%edi)
-	leal	1(%edi), %eax
-	movl	%eax, -32(%ebp)
-	movl	%eax, -68(%ebp)
-.L84:
-	cmpl	-16(%ebp), %ecx
-	jge	.L86
-	movl	-68(%ebp), %eax
-	xorl	%ebx, %ebx
-	addl	%esi, %eax
-	movl	%eax, -60(%ebp)
-.L87:
-	cmpl	%edi, %ebx
-	jge	.L102
-	movl	%ecx, -76(%ebp)
-	movzbl	1052672(%ebx,%esi), %edx
-	incl	%ebx
-	movzbl	1050623(%ebx), %eax
-	call	gf_mul
-	movl	-60(%ebp), %edx
-	movb	%al, 1052671(%ebx,%edx)
-	movl	-76(%ebp), %ecx
-	jmp	.L87
-.L102:
-	movb	1051904(%ecx), %al
-	incl	%ecx
-	movb	%al, 1052673(%esi,%edi,2)
-	addl	-32(%ebp), %esi
-	jmp	.L84
-.L86:
-	movl	-16(%ebp), %eax
-	movl	$1052416, %ecx
-	movl	%edi, %edx
-	call	gauss_solve.constprop.2
-	xorl	%eax, %eax
-.L88:
-	cmpl	%edi, %eax
-	jge	.L71
-	movl	1050880(,%eax,4), %edx
-	incl	%eax
-	movb	1052415(%eax), %cl
-	xorb	%cl, 1052160(%edx)
-	jmp	.L88
-.L73:
-	movb	1052160(%eax), %dl
-	movl	%eax, %ecx
-	incl	%eax
-	movl	-48(%ebp), %ebx
-	sall	$9, %ecx
-	cmpl	-20(%ebp), %eax
-	movb	%dl, (%ebx,%ecx)
-	jl	.L73
-	jmp	.L60
-.L52:
-	addl	$64, %esp
-	popl	%ebx
-	popl	%esi
-	popl	%edi
-	popl	%ebp
-	ret
-	.size	grub_reed_solomon_recover, .-grub_reed_solomon_recover
-	.ident	"GCC: (GNU) 4.8.1"
diff -rupN grub-2.00/util/getroot.c originals/grub-2.00/util/getroot.c
--- grub-2.00/util/getroot.c	2016-03-04 10:32:57.291982167 +0000
+++ originals/grub-2.00/util/getroot.c	2012-06-22 16:42:07.000000000 +0100
@@ -143,13 +143,6 @@
 # endif /* ! RAW_FLOPPY_MAJOR */
 #endif /* defined(__NetBSD__) */
 
-#ifdef __btos__
-#include <sys/limits.h>
-#include <btos.h>
-#include <crt_support.h>
-#define lstat stat
-#endif
-
 #ifdef __linux__
 /* Defines taken from btrfs/ioctl.h.  */
 
@@ -1032,87 +1025,6 @@ grub_find_device (const char *path, dev_
 
 #endif /* __CYGWIN__ */
 
-#ifdef __btos__
-
-static int btos_csv_find_col(const char *csv, const char *col){
-	const char *c = csv;
-	int ret = 0;
-	while(*c != '\0' && *c != '\n'){
-		if(*c == '#' || *c == ','){
-			if(strncmp(c + 2, col, strlen(col)) == 0) return ret;
-			++ret;
-		}
-		++c;
-	}
-	return -1;
-}
-
-static char *btos_get_col(const char *csv, int row, int col){
-	const char *c = csv;
-	int crow = 0;
-	while(crow < row){
-		if(!*c) return NULL;
-		if(*c == '\n') ++crow;
-		++c;
-	}
-	int ccol = 0;
-	while(ccol < col){
-		if(!*c || *c == '\n') return NULL;
-		if(*c == ','){
-			++c;
-			++ccol;
-		}
-		++c;
-	}
-	const char *start = c;
-	int len = 0;
-	while(c && *c && *c != ',' && *c != '\n'){
-		++len;
-		++c;
-	};
-	return strndup(start, len);
-}
-
-static void strtoupper(char *str){
-	while(str && *str){
-		if(*str >= 'a' && *str <= 'z') *str = (*str - 'a') + 'A';
-		++str;
-	}
-}
-
-char *btos_find_device(const char *os_path){
-	char fullpath[BT_MAX_PATH] = {0};
-	btos_path_parse(os_path, fullpath, BT_MAX_PATH);
-	fullpath[btos_get_drive_end(fullpath)] = 0;
-	strtoupper(fullpath);
-
-	bt_handle_t infoh = bt_fopen("info:/mounts", FS_Read);
-	bt_filesize_t size = bt_fseek(infoh, 0, FS_Backwards);
-	bt_fseek(infoh, 0, FS_Set);
-	char *info = (char*)malloc(size);
-	bt_fread(infoh, size, info);
-	bt_fclose(infoh);
-
-	int namecol = btos_csv_find_col(info, "name");
-	int devicecol = btos_csv_find_col(info, "device");
-	int row = 1;
-	char *ret = NULL;
-	while(1){
-		char *name = btos_get_col(info, row, namecol);
-		if(!name) break;
-		if(strcmp(name, fullpath) == 0){
-			ret = btos_get_col(info, row, devicecol);
-		}
-		free(name);
-		if(ret) break;
-		++row;
-	}
-	free(info);
-	return ret;
-}
-
-#endif
-
 char **
 grub_guess_root_devices (const char *dir)
 {
@@ -1182,9 +1094,6 @@ grub_guess_root_devices (const char *dir
   /* GNU/Hurd specific function.  */
   os_dev[0] = find_hurd_root_device (dir);
 
-#elif defined(__btos__)
-  os_dev[0] = btos_find_device (dir);
-
 #else
 
   /* This might be truly slow, but is there any better way?  */
@@ -2185,12 +2094,6 @@ devmapper_out:
       *ptr = 0;
     }
   return ret;
-#elif defined(__btos__)
-  char *ret = xstrdup(os_dev);
-  int ata = (grub_memcmp(ret, "dev:/ata", strlen("dev:/ata")) == 0) && strlen(ret) == strlen("dev:/ataxpy");
-  if(!ata) return ret;
-  ret[9] = 0;
-  return ret;
 #else
 # warning "The function `convert_system_partition_to_system_disk' might not work on your OS correctly."
   return xstrdup (os_dev);
@@ -2472,10 +2375,6 @@ grub_util_biosdisk_get_grub_dev (const c
 
     return make_device_name (drive, strtol (p, NULL, 10) - 1, -1);
   }
-#elif defined(__btos__)
-  {
-    return make_device_name (drive, -1, -1);
-  }
 #else
 # warning "The function `grub_util_biosdisk_get_grub_dev' might not work on your OS correctly."
   return make_device_name (drive, -1, -1);
diff -rupN grub-2.00/util/grub-mkimage.c originals/grub-2.00/util/grub-mkimage.c
--- grub-2.00/util/grub-mkimage.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/util/grub-mkimage.c	2012-06-26 03:15:15.000000000 +0100
@@ -575,7 +575,7 @@ static ISzAlloc g_Alloc = { SzAlloc, SzF
 
 static void
 compress_kernel_lzma (char *kernel_img, size_t kernel_size,
-		      char **core_img, grub_size_t *core_size)
+		      char **core_img, size_t *core_size)
 {
   CLzmaEncProps props;
   unsigned char out_props[5];
@@ -591,7 +591,7 @@ compress_kernel_lzma (char *kernel_img,
   *core_img = xmalloc (kernel_size);
 
   *core_size = kernel_size;
-  if (LzmaEncode ((unsigned char *) *core_img, (SizeT*)core_size,
+  if (LzmaEncode ((unsigned char *) *core_img, core_size,
 		  (unsigned char *) kernel_img,
 		  kernel_size,
 		  &props, out_props, &out_props_size,
@@ -651,7 +651,7 @@ compress_kernel_xz (char *kernel_img, si
 
 static void
 compress_kernel (struct image_target_desc *image_target, char *kernel_img,
-		 size_t kernel_size, char **core_img, grub_size_t *core_size,
+		 size_t kernel_size, char **core_img, size_t *core_size,
 		 grub_compression_t comp)
 {
   if (image_target->flags & PLATFORM_FLAGS_DECOMPRESSORS
@@ -706,7 +706,7 @@ generate_image (const char *dir, const c
 		grub_compression_t comp)
 {
   char *kernel_img, *core_img;
-  grub_size_t kernel_size, total_module_size, core_size, exec_size;
+  size_t kernel_size, total_module_size, core_size, exec_size;
   size_t memdisk_size = 0, config_size = 0, config_size_pure = 0;
   size_t prefix_size = 0;
   char *kernel_path;
diff -rupN grub-2.00/util/misc.c originals/grub-2.00/util/misc.c
--- grub-2.00/util/misc.c	2016-03-02 19:12:57.350274671 +0000
+++ originals/grub-2.00/util/misc.c	2012-03-08 12:49:11.000000000 +0000
@@ -149,7 +149,7 @@ grub_util_write_image_at (const void *im
 			  const char *name)
 {
   grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes at offset 0x%llx",
-		  (unsigned)size, (unsigned long long) offset);
+		  size, (unsigned long long) offset);
   if (fseeko (out, offset, SEEK_SET) == -1)
     grub_util_error (_("cannot seek `%s': %s"),
 		     name, strerror (errno));
@@ -162,7 +162,7 @@ void
 grub_util_write_image (const char *img, size_t size, FILE *out,
 		       const char *name)
 {
-  grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes", (unsigned)size);
+  grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes", size);
   if (fwrite (img, 1, size, out) != size)
     {
       if (!name)
@@ -267,16 +267,6 @@ grub_millisleep (grub_uint32_t ms)
   Sleep (ms);
 }
 
-#elif defined(__btos__)
-
-#include <dev/rtc.h>
-
-void
-grub_millisleep (grub_uint32_t ms)
-{
-  bt_rtc_sleep(ms);
-}
-
 #else
 
 void
