diff -rupN grub-2.00/build-aux/config.sub originals/grub-2.00/build-aux/config.sub
--- grub-2.00/build-aux/config.sub	2017-06-21 14:13:58.271356131 +0100
+++ originals/grub-2.00/build-aux/config.sub	2012-06-18 22:08:46.000000000 +0100
@@ -1368,8 +1368,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-              | -btos* )
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN grub-2.00/configure.ac originals/grub-2.00/configure.ac
--- grub-2.00/configure.ac	2017-06-21 14:18:14.419070842 +0100
+++ originals/grub-2.00/configure.ac	2012-06-26 02:47:37.000000000 +0100
@@ -390,6 +390,7 @@ CC="$TARGET_CC"
 CFLAGS="$TARGET_CFLAGS"
 CPPFLAGS="$TARGET_CPPFLAGS"
 LDFLAGS="$TARGET_LDFLAGS"
+LIBS=""
 
 # debug flags.
 WARN_FLAGS="-Wall -W -Wshadow -Wold-style-definition -Wpointer-arith -Wundef -Wextra -Waddress -Wattributes -Wcast-align -Wchar-subscripts -Wcomment -Wdeprecated-declarations -Wdisabled-optimization -Wdiv-by-zero -Wempty-body -Wendif-labels -Wfloat-equal -Wformat-extra-args -Wformat-security -Wformat-y2k -Wimplicit -Wimplicit-function-declaration -Wimplicit-int -Winit-self -Wint-to-pointer-cast -Winvalid-pch -Wmain -Wmissing-braces -Wmissing-field-initializers -Wmissing-format-attribute -Wmissing-noreturn -Wmultichar -Wnonnull -Woverflow -Wparentheses -Wpointer-arith -Wpointer-to-int-cast -Wreturn-type -Wsequence-point -Wshadow -Wsign-compare -Wstrict-aliasing -Wswitch -Wtrigraphs -Wundef -Wunknown-pragmas -Wunused -Wunused-function -Wunused-label -Wunused-parameter -Wunused-value  -Wunused-variable -Wvariadic-macros -Wvolatile-register-var -Wwrite-strings -Wnested-externs -Wstrict-prototypes -Wpointer-sign"
@@ -689,6 +690,7 @@ CFLAGS="$TARGET_CFLAGS -nostdlib"
 else
 CFLAGS="$TARGET_CFLAGS -nostdlib -Wl,--defsym,___main=0x8100"
 fi
+LIBS=""
 
 # Defined in aclocal.m4.
 grub_PROG_TARGET_CC
@@ -1074,6 +1076,7 @@ fi
 AC_SUBST([LIBZFS])
 AC_SUBST([LIBNVPAIR])
 
+LIBS=""
 
 AC_SUBST([FONT_SOURCE])
 AS_IF([test x$target_cpu = xi386 -a x$platform = xqemu],
diff -rupN grub-2.00/grub-core/disk/ahci.c originals/grub-2.00/grub-core/disk/ahci.c
--- grub-2.00/grub-core/disk/ahci.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/disk/ahci.c	2012-03-10 12:23:16.000000000 +0000
@@ -600,8 +600,8 @@ grub_ahci_readwrite_real (struct grub_ah
 		dev->command_table[0].prdt[0].data_base,
 		dev->command_table[0].prdt[0].unused,
 		dev->command_table[0].prdt[0].size,
-		(unsigned)((char *) &dev->command_table[0].prdt[0]
-		- (char *) &dev->command_table[0]));
+		(char *) &dev->command_table[0].prdt[0]
+		- (char *) &dev->command_table[0]);
 
   if (parms->write)
     grub_memcpy ((char *) grub_dma_get_virt (bufc), parms->buffer, parms->size);
diff -rupN grub-2.00/grub-core/gnulib/sleep.c originals/grub-2.00/grub-core/gnulib/sleep.c
--- grub-2.00/grub-core/gnulib/sleep.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/gnulib/sleep.c	2010-12-01 14:45:43.000000000 +0000
@@ -68,15 +68,6 @@ rpl_sleep (unsigned int seconds)
   return sleep (seconds);
 }
 
-#elif defined(__btos__)
-
-#include <dev/rtc.h>
-
-unsigned int sleep (unsigned int seconds){
-	bt_rtc_sleep(seconds * 1000);
-	return 0;
-}
-
 #else /* !HAVE_SLEEP */
 
  #error "Please port gnulib sleep.c to your platform, possibly using usleep() or select(), then report this to bug-gnulib."
diff -rupN grub-2.00/grub-core/kern/emu/hostdisk.c originals/grub-2.00/grub-core/kern/emu/hostdisk.c
--- grub-2.00/grub-core/kern/emu/hostdisk.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/kern/emu/hostdisk.c	2012-06-25 09:32:04.000000000 +0100
@@ -115,11 +115,6 @@ struct hd_geometry
 # endif /* ! RAW_FLOPPY_MAJOR */
 #endif /* defined(__NetBSD__) */
 
-#if defined(__btos__)
-#include <btos.h>
-#include <crt_support.h>
-#endif
-
 static struct
 {
   char *drive;
@@ -252,7 +247,6 @@ grub_util_get_fd_size (int fd, const cha
   struct disklabel label;
 # elif defined (__sun__)
   struct dk_minfo minfo;
-# elif defined (__btos__)
 # else
   unsigned long long nr;
 # endif
@@ -327,20 +321,6 @@ grub_util_get_fd_size (int fd, const cha
 
  fail:
 
-#elif defined(__btos__)
-
-  sector_size = 512;
-  log_sector_size = 9;
-
-  if (log_secsize) *log_secsize = 9;
-
-  bt_handle_t fh = btos_get_handle(fd);
-  bt_filesize_t pos = bt_fseek(fh, 0, FS_Relative);
-  bt_filesize_t size = bt_fseek(fh, 0, FS_Backwards);
-  bt_fseek(fh, pos, FS_Set);
-
-  return size;
-
   /* In GNU/Hurd, stat() will return the right size.  */
 #elif !defined (__GNU__)
 # warning "No special routine to get the size of a block device is implemented for your OS. This is not possibly fatal."
@@ -799,22 +779,6 @@ grub_util_fd_seek (int fd, const char *n
 		       name, strerror (errno));
   return GRUB_ERR_NONE;
 }
-#elif defined(__btos__)
-grub_err_t
-grub_util_fd_seek (int fd, const char *name, grub_uint64_t off)
-{
-  bt_filesize_t offset = (bt_filesize_t) off;
-  bt_handle_t handle = btos_get_handle(fd);
-  grub_dprintf ("hostdisk", "seeking %x%x on handle %d\n", (uint32_t)(offset >> 32), (uint32_t)offset, (int)handle);
-  bt_filesize_t actual = bt_fseek(handle, offset, FS_Set);
-
-  if (actual != offset){
-    grub_dprintf ("hostdisk", "seek failed! (%x%x != %x%x)\n", (uint32_t)(actual >> 32), (uint32_t)actual, (uint32_t)(offset >> 32), (uint32_t)offset);
-    return grub_error (GRUB_ERR_BAD_DEVICE, N_("cannot seek `%s': %s"),
-		       name, strerror (errno));
-  }
-  return 0;
-}
 #else
 grub_err_t
 grub_util_fd_seek (int fd, const char *name, grub_uint64_t off)
diff -rupN grub-2.00/grub-core/kern/emu/mm.c originals/grub-2.00/grub-core/kern/emu/mm.c
--- grub-2.00/grub-core/kern/emu/mm.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/kern/emu/mm.c	2012-02-07 23:13:14.000000000 +0000
@@ -25,10 +25,6 @@
 #include <string.h>
 #include <grub/i18n.h>
 
-#ifdef __btos__
-#include <malloc.h>
-#endif
-
 void *
 grub_malloc (grub_size_t size)
 {
diff -rupN grub-2.00/grub-core/lib/backtrace.c originals/grub-2.00/grub-core/lib/backtrace.c
--- grub-2.00/grub-core/lib/backtrace.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/lib/backtrace.c	2012-02-26 19:56:25.000000000 +0000
@@ -39,7 +39,7 @@ grub_backtrace_print_address (void *addr
 	  + segment->size > (grub_uint8_t *) addr)
 	{
 	  grub_printf ("%s.%x+%" PRIxGRUB_SIZE, mod->name, segment->section,
-		       (unsigned)((grub_uint8_t *) addr - (grub_uint8_t *) segment->addr));
+		       (grub_uint8_t *) addr - (grub_uint8_t *) segment->addr);
 	  return;
 	}
   }
diff -rupN grub-2.00/grub-core/lib/minilzo/lzoconf.h originals/grub-2.00/grub-core/lib/minilzo/lzoconf.h
--- grub-2.00/grub-core/lib/minilzo/lzoconf.h	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/lib/minilzo/lzoconf.h	2011-10-14 18:10:35.000000000 +0100
@@ -53,7 +53,7 @@
 #  include <config.h>
 #endif
 #include <limits.h>
-#include <stdint.h>
+#include <stddef.h>
 
 
 /***********************************************************************
diff -rupN grub-2.00/grub-core/lib/minilzo/minilzo.c originals/grub-2.00/grub-core/lib/minilzo/minilzo.c
--- grub-2.00/grub-core/lib/minilzo/minilzo.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/lib/minilzo/minilzo.c	2012-02-29 14:04:03.000000000 +0000
@@ -58,12 +58,7 @@
 #  include <config.h>
 #endif
 #include <limits.h>
-#include <stdint.h>
-#include <sys/types.h>
-#ifndef NULL
-#define NULL 0
-#endif
-typedef __PTRDIFF_TYPE__ ptrdiff_t;
+#include <stddef.h>
 #if defined(MINILZO_CFG_USE_INTERNAL_LZODEFS)
 
 #ifndef __LZODEFS_H_INCLUDED
diff -rupN grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c originals/grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c
--- grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/lib/xzembed/xz_dec_lzma2.c	2010-12-01 14:45:43.000000000 +0000
@@ -24,8 +24,6 @@
 #include "xz_private.h"
 #include "xz_lzma2.h"
 
-#pragma GCC diagnostic ignored "-Wattributes"
-
 /*
  * Range decoder initialization eats the first five bytes of each LZMA chunk.
  */
diff -rupN grub-2.00/grub-core/net/ip.c originals/grub-2.00/grub-core/net/ip.c
--- grub-2.00/grub-core/net/ip.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/net/ip.c	2012-06-09 10:12:46.000000000 +0100
@@ -403,7 +403,7 @@ grub_net_recv_ip4_packets (struct grub_n
 					    * sizeof (grub_uint32_t)))
     {
       grub_dprintf ("net", "IP packet too short: %" PRIdGRUB_SSIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    (nb->tail - nb->data));
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
@@ -654,7 +654,7 @@ grub_net_recv_ip6_packets (struct grub_n
   if (nb->tail - nb->data < (grub_ssize_t) sizeof (*iph))
     {
       grub_dprintf ("net", "IP packet too short: %" PRIdGRUB_SSIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/net/tcp.c originals/grub-2.00/grub-core/net/tcp.c
--- grub-2.00/grub-core/net/tcp.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/net/tcp.c	2012-06-22 16:42:07.000000000 +0100
@@ -766,7 +766,7 @@ grub_net_recv_tcp_packet (struct grub_ne
 					     >> 12) * sizeof (grub_uint32_t)))
     {
       grub_dprintf ("net", "TCP packet too short: %" PRIuGRUB_SIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/net/udp.c originals/grub-2.00/grub-core/net/udp.c
--- grub-2.00/grub-core/net/udp.c	2017-06-21 14:13:58.607355757 +0100
+++ originals/grub-2.00/grub-core/net/udp.c	2012-06-09 10:12:46.000000000 +0100
@@ -155,7 +155,7 @@ grub_net_recv_udp_packet (struct grub_ne
   if (nb->tail - nb->data < (grub_ssize_t) sizeof (*udph))
     {
       grub_dprintf ("net", "UDP packet too short: %" PRIuGRUB_SIZE "\n",
-		    (int)(nb->tail - nb->data));
+		    nb->tail - nb->data);
       grub_netbuff_free (nb);
       return GRUB_ERR_NONE;
     }
diff -rupN grub-2.00/grub-core/script/yylex.l originals/grub-2.00/grub-core/script/yylex.l
--- grub-2.00/grub-core/script/yylex.l	2017-06-21 14:13:58.619355744 +0100
+++ originals/grub-2.00/grub-core/script/yylex.l	2012-06-08 21:24:15.000000000 +0100
@@ -29,8 +29,6 @@
 #pragma GCC diagnostic ignored "-Wmissing-prototypes"
 #pragma GCC diagnostic ignored "-Wmissing-declarations"
 #pragma GCC diagnostic ignored "-Wunsafe-loop-optimizations"
-#pragma GCC diagnostic ignored "-Wunused-function"
-#pragma GCC diagnostic ignored "-Wsign-compare"
 
 #define yyfree    grub_lexer_yyfree
 #define yyalloc   grub_lexer_yyalloc
diff -rupN grub-2.00/util/getroot.c originals/grub-2.00/util/getroot.c
--- grub-2.00/util/getroot.c	2017-06-21 14:13:58.623355740 +0100
+++ originals/grub-2.00/util/getroot.c	2012-06-22 16:42:07.000000000 +0100
@@ -143,13 +143,6 @@
 # endif /* ! RAW_FLOPPY_MAJOR */
 #endif /* defined(__NetBSD__) */
 
-#ifdef __btos__
-#include <sys/limits.h>
-#include <btos.h>
-#include <crt_support.h>
-#define lstat stat
-#endif
-
 #ifdef __linux__
 /* Defines taken from btrfs/ioctl.h.  */
 
@@ -1032,87 +1025,6 @@ grub_find_device (const char *path, dev_
 
 #endif /* __CYGWIN__ */
 
-#ifdef __btos__
-
-static int btos_csv_find_col(const char *csv, const char *col){
-	const char *c = csv;
-	int ret = 0;
-	while(*c != '\0' && *c != '\n'){
-		if(*c == '#' || *c == ','){
-			if(strncmp(c + 2, col, strlen(col)) == 0) return ret;
-			++ret;
-		}
-		++c;
-	}
-	return -1;
-}
-
-static char *btos_get_col(const char *csv, int row, int col){
-	const char *c = csv;
-	int crow = 0;
-	while(crow < row){
-		if(!*c) return NULL;
-		if(*c == '\n') ++crow;
-		++c;
-	}
-	int ccol = 0;
-	while(ccol < col){
-		if(!*c || *c == '\n') return NULL;
-		if(*c == ','){
-			++c;
-			++ccol;
-		}
-		++c;
-	}
-	const char *start = c;
-	int len = 0;
-	while(c && *c && *c != ',' && *c != '\n'){
-		++len;
-		++c;
-	};
-	return strndup(start, len);
-}
-
-static void strtoupper(char *str){
-	while(str && *str){
-		if(*str >= 'a' && *str <= 'z') *str = (*str - 'a') + 'A';
-		++str;
-	}
-}
-
-char *btos_find_device(const char *os_path){
-	char fullpath[BT_MAX_PATH] = {0};
-	btos_path_parse(os_path, fullpath, BT_MAX_PATH);
-	fullpath[btos_get_drive_end(fullpath)] = 0;
-	strtoupper(fullpath);
-
-	bt_handle_t infoh = bt_fopen("info:/mounts", FS_Read);
-	bt_filesize_t size = bt_fseek(infoh, 0, FS_Backwards);
-	bt_fseek(infoh, 0, FS_Set);
-	char *info = (char*)malloc(size);
-	bt_fread(infoh, size, info);
-	bt_fclose(infoh);
-
-	int namecol = btos_csv_find_col(info, "name");
-	int devicecol = btos_csv_find_col(info, "device");
-	int row = 1;
-	char *ret = NULL;
-	while(1){
-		char *name = btos_get_col(info, row, namecol);
-		if(!name) break;
-		if(strcmp(name, fullpath) == 0){
-			ret = btos_get_col(info, row, devicecol);
-		}
-		free(name);
-		if(ret) break;
-		++row;
-	}
-	free(info);
-	return ret;
-}
-
-#endif
-
 char **
 grub_guess_root_devices (const char *dir)
 {
@@ -1182,9 +1094,6 @@ grub_guess_root_devices (const char *dir
   /* GNU/Hurd specific function.  */
   os_dev[0] = find_hurd_root_device (dir);
 
-#elif defined(__btos__)
-  os_dev[0] = btos_find_device (dir);
-
 #else
 
   /* This might be truly slow, but is there any better way?  */
@@ -2185,16 +2094,6 @@ devmapper_out:
       *ptr = 0;
     }
   return ret;
-#elif defined(__btos__)
-  char *ret = xstrdup(os_dev);
-  char *c;
-  for(c = ret; *c; ++c){
-     if(*c >= 'A' && *c <= 'Z') *c = (*c - 'A') + 'a';
-  }
-  int ata = (grub_memcmp(ret, "dev:/ata", strlen("dev:/ata")) == 0) && strlen(ret) == strlen("dev:/ataxpy");
-  if(!ata) return ret;
-  ret[9] = 0;
-  return ret;
 #else
 # warning "The function `convert_system_partition_to_system_disk' might not work on your OS correctly."
   return xstrdup (os_dev);
@@ -2476,10 +2375,6 @@ grub_util_biosdisk_get_grub_dev (const c
 
     return make_device_name (drive, strtol (p, NULL, 10) - 1, -1);
   }
-#elif defined(__btos__)
-  {
-    return make_device_name (drive, -1, -1);
-  }
 #else
 # warning "The function `grub_util_biosdisk_get_grub_dev' might not work on your OS correctly."
   return make_device_name (drive, -1, -1);
diff -rupN grub-2.00/util/grub-mkimage.c originals/grub-2.00/util/grub-mkimage.c
--- grub-2.00/util/grub-mkimage.c	2017-06-21 14:13:58.623355740 +0100
+++ originals/grub-2.00/util/grub-mkimage.c	2012-06-26 03:15:15.000000000 +0100
@@ -575,7 +575,7 @@ static ISzAlloc g_Alloc = { SzAlloc, SzF
 
 static void
 compress_kernel_lzma (char *kernel_img, size_t kernel_size,
-		      char **core_img, grub_size_t *core_size)
+		      char **core_img, size_t *core_size)
 {
   CLzmaEncProps props;
   unsigned char out_props[5];
@@ -591,7 +591,7 @@ compress_kernel_lzma (char *kernel_img,
   *core_img = xmalloc (kernel_size);
 
   *core_size = kernel_size;
-  if (LzmaEncode ((unsigned char *) *core_img, (SizeT*)core_size,
+  if (LzmaEncode ((unsigned char *) *core_img, core_size,
 		  (unsigned char *) kernel_img,
 		  kernel_size,
 		  &props, out_props, &out_props_size,
@@ -651,7 +651,7 @@ compress_kernel_xz (char *kernel_img, si
 
 static void
 compress_kernel (struct image_target_desc *image_target, char *kernel_img,
-		 size_t kernel_size, char **core_img, grub_size_t *core_size,
+		 size_t kernel_size, char **core_img, size_t *core_size,
 		 grub_compression_t comp)
 {
   if (image_target->flags & PLATFORM_FLAGS_DECOMPRESSORS
@@ -706,7 +706,7 @@ generate_image (const char *dir, const c
 		grub_compression_t comp)
 {
   char *kernel_img, *core_img;
-  grub_size_t kernel_size, total_module_size, core_size, exec_size;
+  size_t kernel_size, total_module_size, core_size, exec_size;
   size_t memdisk_size = 0, config_size = 0, config_size_pure = 0;
   size_t prefix_size = 0;
   char *kernel_path;
diff -rupN grub-2.00/util/misc.c originals/grub-2.00/util/misc.c
--- grub-2.00/util/misc.c	2017-06-21 14:13:58.627355736 +0100
+++ originals/grub-2.00/util/misc.c	2012-03-08 12:49:11.000000000 +0000
@@ -149,7 +149,7 @@ grub_util_write_image_at (const void *im
 			  const char *name)
 {
   grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes at offset 0x%llx",
-		  (unsigned)size, (unsigned long long) offset);
+		  size, (unsigned long long) offset);
   if (fseeko (out, offset, SEEK_SET) == -1)
     grub_util_error (_("cannot seek `%s': %s"),
 		     name, strerror (errno));
@@ -162,7 +162,7 @@ void
 grub_util_write_image (const char *img, size_t size, FILE *out,
 		       const char *name)
 {
-  grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes", (unsigned)size);
+  grub_util_info ("writing 0x%" PRIxGRUB_SIZE " bytes", size);
   if (fwrite (img, 1, size, out) != size)
     {
       if (!name)
@@ -267,16 +267,6 @@ grub_millisleep (grub_uint32_t ms)
   Sleep (ms);
 }
 
-#elif defined(__btos__)
-
-#include <dev/rtc.h>
-
-void
-grub_millisleep (grub_uint32_t ms)
-{
-  bt_rtc_sleep(ms);
-}
-
 #else
 
 void
